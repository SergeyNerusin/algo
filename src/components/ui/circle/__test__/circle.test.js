import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ElementStates } from '../../../../types/element-states';

import { Circle } from '../circle';

describe('testing Circle component', () => {
  test('circle without text render correctly', () => {
    const button = TestRenderer.create(<Circle letter='' />).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('circle with text render correctly', () => {
    const button = TestRenderer.create(<Circle letter='123' />).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('circle render correctly with "head"', () => {
    const button = TestRenderer.create(
      <Circle letter='' head='head' />
    ).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('circle render correctly with "head" and letter', () => {
    const button = TestRenderer.create(
      <Circle letter='123' head='head' />
    ).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('circle render correctly with "tail"', () => {
    const button = TestRenderer.create(
      <Circle letter='' tail='tail' />
    ).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('circle render correctly with "tail" and letter', () => {
    const button = TestRenderer.create(
      <Circle letter='123' tail='tail' />
    ).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('circle with index render correctly', () => {
    const button = TestRenderer.create(<Circle index={0} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('circle with isSmall render correctly', () => {
    const button = TestRenderer.create(<Circle isSmall={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('circle with state chainging render correctly', () => {
    const button = TestRenderer.create(
      <Circle state={ElementStates.Changing} />
    ).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('circle with state modified render correctly', () => {
    const button = TestRenderer.create(
      <Circle state={ElementStates.Modified} />
    ).toJSON();
    expect(button).toMatchSnapshot();
  });
});
