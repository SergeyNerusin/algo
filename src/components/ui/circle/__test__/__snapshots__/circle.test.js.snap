// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Circle component circle render correctly with "head" 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component circle render correctly with "head" and letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component circle render correctly with "tail" 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`testing Circle component circle render correctly with "tail" and letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`testing Circle component circle with index render correctly 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`testing Circle component circle with isSmall render correctly 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component circle with state chainging render correctly 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component circle with state modified render correctly 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component circle with text render correctly 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      123
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`testing Circle component circle without text render correctly 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
